contents/tutorials/track-high-volume-apis.md

```diff
@@ -0,0 +1,266 @@
+
+1. Sampling using feature flags
+2. Caching usage and batching capture requests
+
+After doing this, we also show the different ways you can calculate insights.
+
+## Building our API
+
+To start, we will build a basic API that captures and returns the ID we passed to it. We use [Express, the Node server framework](/tutorials/node-express-analytics), to do this
+
+Create a folder for our project, go into the folder, and initialize npm. When prompted, select all the default options.
+
+```bash
+mkdir high-volume
+cd high-volume
+npm init
+```
+
+Once done, install Express.
+
+```bash
+npm i express
+```
+
+In your `high-volume` folder, create a file named `index.js`, and set up a basic Express app.
+
+```node
+// index.js
+const express = require('express');
+const app = express();
+const port = 3000
+
+app.get('/', (req, res) => {
+  res.send('Hello World!');
+})
+
+app.listen(port, () => {
+  console.log(`App listening on http://localhost:${port}`)
+})
+```
+
+Lastly, create one more route as our high volume requests route. We set this route as  `/big/:id`. For now, it returns a message using the request ID.
+
+```node
+//...
+app.get('/big/:id', (req, res) => {
+  res.send(`Welcome to the big route, ${req.params.id}`);
+})
+//...
+```
+
+### Adding PostHog
+
+Next, we install and set up PostHog. 
+
+```bash
+npm i posthog-node
+```
+
+We set up the PostHog client with our project API key and instance address, both of which you can get from your [project settings](https://app.posthog.com/project/settings). We add a `client.capture()` call to our `/big` route. The capture call includes the route distinct ID and an event name. We also include a `firstRouteCalled` user property to use for a filter later. 
+
+```node
+const express = require('express');
+const { PostHog } = require('posthog-node');
+const app = express();
+const port = 3000
+
+const client = new PostHog(
+  '<ph_project_api_key>',
+  { host: '<ph_instance_address>'}
+)
+
+//...
+
+app.get('/big/:id', (req, res) => {
+  client.capture(
+    { 
+      event: 'big route',
+      distinctId: req.params.id,
+			properties: {
+        $set_once: { firstRouteCalled: req.url },   
+      },
+    },
+  );
+  res.send(`Welcome to the big route, ${req.params.id}`);
+})
+//...
+```
+
+When we go to the big route again, we capture an event that we can see in our PostHog instance.
+
+![Event](../images/tutorials/track-high-volume-apis/event.png)
+
+In our situation, capturing every single request to this route creates many requests. It hurts the performance of our API to make a capture request every time. Instead, we will use two different strategies for capturing them: sampling and batching.
+
+## 1. Sampling with feature flags and local evaluation
+
+We can use feature flags to sample users. This way we can easily control what amount of events we capture. For example, we can set up a flag to only capture specific or a percentage of:
```

Comment:
Alternatively also: `This way, we can easily control the quantity of events that we capture`
```suggestion
We can use feature flags to sample users. This way, we can easily control how many events we capture. For example, we can set up a flag to only capture specific or a percentage of:
```

Reply:
Also, perhaps worth mentioning another benefit is that you can control this without having to redeploy a new build and can just update the number in the dashboard?